### PR TITLE
BAU: Set correct intermediate metadata cert validity period

### PR DESCRIPTION
**Pull request by CharlesIC**
**Set correct intermediate metadata cert validity period**
Intermediate metadata certs in prod should have a validity of 5 years, the same as our current certs. SAML signing certs are rotated every 3 months.

**Remove filtering out of leaf metadata signing cert from the published certs**
This is no longer necessary as the VMC has been modified to not include the leaf certs in the CA certs secret. Remove caching of certs view as CA certs can now be mounted dynamically by the K8s cluster.

### CODE REVIEW

**Review by idrop — DISMISSED** [review]

**Review by idrop — APPROVED** [review]
"Appease Snyk" 👍

File: chart/templates/metadata.yaml [hunk 1]
```diff
@@ -19,11 +19,12 @@ spec:
     contactGivenName: GOV.UK Verify
     contactSurname: Support
     contactEmail: idasupport@digital.cabinet-office.gov.uk
+    validityDays: 14
   enabled: true
   samlSigningCertRequest:
     countryCode: GB
     commonName: Verify Proxy Node SAML Signing
-    expiryMonths: 2
+    expiryMonths: 3
```
Comment (Wynndow) [inline chart/templates/metadata.yaml]: Shouldn't this be 9 months? The plan is they expire after 9 months but we rotate after 3, right?

Comment (CharlesIC) [reply chart/templates/metadata.yaml]: I guess but if we plan to rotate them every 3 months then it's effectively the same. We can change this to 9 months if needed when we implement the cert rotation in VMC

File: ci/build/build-pipeline.yaml [hunk 2]
```diff
@@ -262,6 +262,7 @@ spec:
           trigger: true
 
       - task: run-tests
+        attempts: 2
```
Comment (JohnWattsGDS) [inline ci/build/build-pipeline.yaml]: Out of curiosity; why are we now attempting the tests twice?

**Review by Wynndow — APPROVED** [review]

File: ci/build/build-pipeline.yaml (hunk 2 above)
Comment (CharlesIC) [reply ci/build/build-pipeline.yaml]: Cos they randomly fail sometimes 🤷‍♂

Comment (JohnWattsGDS) [reply ci/build/build-pipeline.yaml]: As good a reason as any!

Comment (paroxp) [reply ci/build/build-pipeline.yaml]: 😬

**Pull request merged**